### DISCUSSION TITLE
Add Integration for SQL Querying

### DISCUSSION THREAD

**Discussion by smargoli2** [tryAGI/LangChain#3] [Ideas]
Hey,

I'm looking for a C# replacement for the langchain sql_database_agent which can convert natural language into sql queries and query my sql server database. Any ideas on how to implement?

**Comment by HavenDV** [comment]
Hello. Sorry, the project is unusable in its current state. There is still a lot to be done to make sql_database_agent work.

**Reply by smargoli2** [reply]
Any pointers for me to develop that on my own?

**Reply by HavenDV** [reply]
Need to implement following code in C#:
https://python.langchain.com/docs/integrations/toolkits/sql_database
https://github.com/langchain-ai/langchain/tree/master/libs/langchain/langchain/chains/sql_database
https://github.com/langchain-ai/langchain/tree/master/libs/langchain/langchain/agents/agent_toolkits/sql
https://github.com/langchain-ai/langchain/blob/master/libs/experimental/langchain_experimental/sql/

**Reply by lisquisbertp** [reply]
Hello,

I need the langchain sql_database_agent feature as well, do you have some news about it? please
